subprojects/base-services/src/main/java/org/gradle/internal/jvm/Jvm.java

```diff
@@ -63,10 +64,10 @@ public static Jvm current() {
     @VisibleForTesting
     static JvmImplementation createCurrent() {
         String vendor = System.getProperty("java.vm.vendor");
-        if (vendor.toLowerCase().startsWith("apple inc.")) {
+        if (TextUtil.toLowerCaseUserLocale(vendor).startsWith("apple inc.")) {
             return new AppleJvm(OperatingSystem.current());
         }
-        if (vendor.toLowerCase().startsWith("ibm corporation")) {
+        if (TextUtil.toLowerCaseUserLocale(vendor).startsWith("ibm corporation")) {
```

Comment:
I think both of these need to be English locale since we're looking for a specific string.